crawl-ref/source/item-name.cc

```diff
@@ -1952,6 +1952,31 @@ bool item_type_known(const object_class_type base_type, const int sub_type)
     return you.type_ids[base_type][sub_type];
 }
 
+// Has every item a scroll can identify (i.e. all scrolls and potions) been
+// identified by "you"?
+static bool _is_everything_identified;
```

Comment:
Would prefer not to do something like this with a static, and if with a static, with something that ensured a default initialization (e.g. via https://isocpp.org/wiki/faq/ctors#construct-on-first-use-v2). See below for a completely non-static suggestion. I realize you are doing initialization by running the check at startup but it's still better to ensure default initialization absolutely in case the code is run in an unexpected way -- as-is in this patch `_is_everything_identified` has an undefined value until the `check...` function is run. (E.g. I couldn't swear that that startup code is called during tests or scripts, though it might be. Similar with arena.)

Overall, I think it might be simpler, instead of using a static, to set a prop on the player when they have identified everything, and then condition the id uselessness check on that. You wouldn't even need a meaningful value -- just whether it is set. This would let the value get saved, not need a static, and side-step the issues about default initialization (if the check is whether a prop is set, the default is a per-player no, automatically).